crates/driver/src/infra/solver/mod.rs

```diff
@@ -125,7 +125,13 @@ impl Solver {
             .client
             .post(self.config.endpoint.clone())
             .body(body)
-            .timeout(timeout.into());
+            .timeout(
+                // Add a small buffer to the timeout to account for network latency. Otherwise the
+                // HTTP timeout might happen before the solver times out its search algorithm.
+                (timeout.duration() + chrono::Duration::milliseconds(500))
```

Comment:
Do you think 500 milliseconds is the right value? Should it be lower? Or higher?

Reply:
I think 500ms should probably enough. However, I would say we should rather deduct that from the solver deadline and leave the http timeout as is.
My reasoning is that this would give solvers stronger guarantees. For example in case they have to manage inventory risk for their solutions they don't suddenly have to account for additional 500ms they wouldn't really know about. These timing concerns came up a lot while talking to the MooMaker team.

Reply:
Agree with @MartinquaXD